pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/PendingAckHandleImpl.java

```diff
@@ -975,11 +975,29 @@ public TransactionInPendingAckStats getTransactionInPendingAckStats(TxnID txnID)
     }
 
     @Override
-    public CompletableFuture<Void> close() {
+    public CompletableFuture<Void> closeAsync() {
         changeToCloseState();
         synchronized (PendingAckHandleImpl.this) {
             if (this.pendingAckStoreFuture != null) {
-                return this.pendingAckStoreFuture.thenAccept(PendingAckStore::closeAsync);
+                CompletableFuture<Void> closeFuture = new CompletableFuture<>();
+                this.pendingAckStoreFuture.whenComplete((v, e) -> {
+                    if (e != null) {
```

Comment:
Suggest using `pendingAckStore` instead of `v` because I almost thought you forgot to close `PendingAckStore`. After double-checked and found that the `v` is `PendingAckStore`.